python/pyspark/sql/readwriter.py

```diff
@@ -733,7 +733,7 @@ def save(self, path=None, format=None, mode=None, partitionBy=None, **options):
         :param partitionBy: names of partitioning columns
         :param options: all other string options
 
-        >>> df.write.mode('append').parquet(os.path.join(tempfile.mkdtemp(), 'data'))
+        >>> df.select("day", "month", "name", "year").write.save("mydata.parquet")
```

Comment:
I guess we need `mode("overwrite")`.
```
File "/home/jenkins/workspace/SparkPullRequestBuilder/python/pyspark/sql/readwriter.py", line 735, in pyspark.sql.readwriter.DataFrameWriter.save
Failed example:
    df.select("day", "month", "name", "year").write.save("mydata.parquet")
Exception raised:
    Traceback (most recent call last):
      File "/home/anaconda/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest pyspark.sql.readwriter.DataFrameWriter.save[0]>", line 1, in <module>
        df.select("day", "month", "name", "year").write.save("mydata.parquet")
      File "/home/jenkins/workspace/SparkPullRequestBuilder/python/pyspark/sql/readwriter.py", line 746, in save
        self._jwrite.save(path)
      File "/home/jenkins/workspace/SparkPullRequestBuilder/python/lib/py4j-0.10.8.1-src.zip/py4j/java_gateway.py", line 1286, in __call__
        answer, self.gateway_client, self.target_id, self.name)
      File "/home/jenkins/workspace/SparkPullRequestBuilder/python/pyspark/sql/utils.py", line 102, in deco
        raise converted
    AnalysisException: path file:/home/jenkins/workspace/SparkPullRequestBuilder/mydata.parquet already exists.;
```

Reply:
BTW, let's remove `.select("day", "month", "name", "year")`. We don't need to `select` because this is `save` API doc.